src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -568,3 +568,153 @@ def test_lfric_driver_field_array_inc():
+
+    # Even though PSyclone cannot find the variable, it should still be
+    # extracted:
+
+    filename = "driver-import-test.F90"
+    with open(filename, "r", encoding='utf-8') as my_file:
+        driver = my_file.read()
+
+    assert ("call extract_psy_data%ReadVariable("
+            "'f1_data@module_with_name_clash_mod', f1_data_1)" in driver)
+    assert ("call extract_psy_data%ReadVariable("
+            "'f2_data@module_with_name_clash_mod', f2_data_1)" in driver)
+    assert ("call extract_psy_data%ReadVariable("
+            "'f2_data_post@module_with_name_clash_mod', f2_data_1_post)"
+            in driver)
+
+
+# ----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir", "init_module_manager")
+def test_lfric_driver_external_symbols_error(capsys):
+    '''Test the handling of symbols imported from other modules, or calls to
+    external functions that use module variables. In this example, the
+    external module cannot be parsed by fparser (it contains syntax errors),
+    resulting in external functions and variables that cannot be found.
+
+    '''
+    _, invoke = get_invoke("driver_creation/invoke_kernel_with_imported_"
+                           "symbols_error.f90", API, dist_mem=False, idx=0)
+
+    extract = LFRicExtractTrans()
+    extract.apply(invoke.schedule.children[0],
+                  options={"create_driver": True,
+                           "region_name": ("import", "test")})
+    code = str(invoke.gen())
+    # Even though PSyclone cannot find the variable, it should still be
+    # extracted:
+    assert ('CALL extract_psy_data%PreDeclareVariable("non_existent_var@'
+            'module_with_error_mod", non_existent_var' in code)
+    assert ('CALL extract_psy_data%ProvideVariable("non_existent_var@'
+            'module_with_error_mod", non_existent_var' in code)
+
+    filename = "driver-import-test.F90"
+    with open(filename, "r", encoding='utf-8') as my_file:
+        driver = my_file.read()
+
+    # First check output of extraction, which will detect the problems of
+    # finding variables and functions:
+    out, _ = capsys.readouterr()
+    assert ("Cannot find symbol 'non_existent_func' in module "
+            "'module_with_error_mod' - ignored." in out)
+    assert ("Index error finding 'non_existent_var' in "
+            "'module_with_error_mod'." in out)
+
+    # This error comes from the driver creation: a variable is in the list
+    # of variables to be processed, but its type cannot be found.
+    assert ("Cannot find variable with tag 'non_existent_var@module_with_"
+            "error_mod' - likely a symptom of an earlier parsing problem."
+            in out)
+    # This variable will be ignored (for now, see TODO 2120) so no code will
+    # be created for it. The string will still be in the created driver (since
+    # the module is still inlined), but no ReadVariable code should be created:
```

Comment:
What will this mean for the created driver - that there will be uninitialised variables? If so, all the more reason to send the message to stderr to alert the user.

Reply:
Postponed for #2120